libvast/src/system/accountant.cpp

```diff
@@ -134,21 +175,42 @@ accountant_type::behavior_type accountant(accountant_actor* self,
                                           const path& filename) {
   using namespace std::chrono;
   init(self, filename);
-  self->set_exit_handler(
-    [=](const caf::exit_msg& msg) {
-      self->state.file.flush();
-      self->quit(msg.reason);
-    }
-  );
+  self->set_exit_handler([=](const caf::exit_msg& msg) {
+    finish_slice(self);
+    self->state.file.flush();
+    self->quit(msg.reason);
+  });
   self->set_down_handler(
     [=](const caf::down_msg& msg) {
       VAST_DEBUG(self, "received DOWN from", msg.source);
       self->state.actor_map.erase(msg.source.id());
     }
   );
+  self->state.mgr = self->make_continuous_source(
+    // init
+    [=](bool&) {},
+    // get next element
+    [=](bool&, caf::downstream<table_slice_ptr>& out, size_t num) {
+      auto& st = self->state;
+      size_t produced = 0;
+      while (num-- > 0 && !st.slice_buffer.empty()) {
+        auto& slice = st.slice_buffer.front();
+        produced += slice->rows();
+        out.push(std::move(slice));
+        st.slice_buffer.pop();
+      }
+      VAST_TRACE(self, "was asked for", num, "slices and produced", produced,
+                 ".", st.slice_buffer.size(), "are remaining in buffer");
```

Comment:
```suggestion
                 ";", st.slice_buffer.size(), "are remaining in buffer");
```
Since none of our logging messages and with a period, maybe `;` seems like a more natural separator.